core/src/main/resources/filodb-defaults.conf

```diff
@@ -280,7 +280,11 @@ akka {
 
     # Auto downing is turned off by default.  See
     # http://doc.akka.io/docs/akka/2.3.16/scala/cluster-usage.html#Automatic_vs__Manual_Downing
-    # auto-down-unreachable-after = 120s
+    # Instead, we use this repo to provide smarter downing solutions:
+    # https://github.com/TanUkkii007/akka-cluster-custom-downing
+    # Note that this strategy should work well but for fixed size clusters QuorumLeader might be better
+    downing-provider-class = "tanukki.akka.cluster.autodown.MajorityLeaderAutoDowning"
```

Comment:
How is "majority" count of the cluster mutated? In other words, what if some nodes go down and never come up because of an infrastructure issue - how long does it take for majority to be updated ?

If there is a split brain before the majority is updated, the entire cluster could go down ?

Reply:
The "majority" algorithm is not foolproof, it is not meant to be the production strategy, but a better strategy than auto-downing for dev environments.   For production we'd use the `StaticQuorum` strategy which is pretty foolproof for static cluster sizes.   I just wanted a default which was easy to configure.

Here is the code for the MajorityLeader thing though.

https://github.com/TanUkkii007/akka-cluster-custom-downing/blob/master/src/main/scala/tanukki/akka/cluster/autodown/MajorityAwareCustomAutoDownBase.scala

Reply:
Quorum is also not foolproof in the exact same scenario. A few nodes go down, and don't come back up. We then see a split brain with fewer than quorum number of nodes on both sides. Both sides are brought down.

I actually like the MajorityLeader better since it would tolerate this scenario better, but only if the "majority" is updated reliably on both split brain sides.

Reply:
Is this the scenario you are saying?
a) A few nodes go down... in which case, the majority is still up.  Both algorithms should work
b) A few nodes go down and the remaining nodes are split by a partition. In which case there is no majority.

In b), the Quorum would still work in the sense that both sides go down.  This is fine, there is still no split brain.  A whole new cluster will be created by the cloud provider.

Reply:
In (b) we _do not_ want all nodes to go down. Only one side of the split brain should be brought down. Why rebuild the entire cluster unnecessarily ?

Reply:
If both are minorities, then there is no automatic way to choose which side is right, so some kind of manual intervention is required.  I think it is better for all nodes to go down, thus new nodes will come up, then to have a temporary split brain.  
However I think this is extremely unlikely.